### ISSUE TITLE
Allow to go back to the previous state

### ISSUE THREAD

**Issue by uesteibar** [retro-tool/retro-tool#41]

**Comment by uesteibar** [comment]
I think this is fairly critical because not everybody understand the _steps_ currently, so a way to fix mistakes would be welcome.

What do you think @retro-tool/all ?

**Comment by marciobarrios** [comment]
I would try to avoid this feature, I would explore alternatives to better explain the process (the landing is an improvement that will help with this topic), I think we all agree on reducing the options and keep the functionality to a minimum to promote simplicity versus other retrospective tools.

**Comment by uesteibar** [comment]
still, I did have to get into the database right now to move one retro back to the `review` state, so I think this would be really useful for some folks.

**Comment by rudeayelo** [comment]
I don't think it's necessary either, it even has a modal to ensure you understand you can't go back ffs.

I'd add it if we start receiving more feedback on that request, no big deal anyway.

**Comment by uesteibar** [comment]
The modal only appears the first time you jump phase, right? Maybe we can add it for all jumps? And include some explanation of the next step.

In the case that prompted me to open this issue, the person that did jump to the next phase didn't even notice they did. Maybe always showing a modal would prevent that?

**Comment by rudeayelo** [comment]
I think there's already some checks preventing that:

- **Add items:** If there's no items, the button to start the retro is disabled, if they exist, then we show the modal
- **Group and vote:** If no one voted, the button is disabled, maybe it's not clear why, but at least we prevent skipping this phase by mistake.

Things that we could do additionally:

- a tooltip in the button explaining why it's disabled
- also disabling the button in the **Add action items phase** until some action item is added, some of our retros end without action items, so maybe we could add another modal here in case no action items were filled:
![image](https://user-images.githubusercontent.com/140042/82016031-6902e880-9680-11ea-894a-e728b47379b4.png)

**Comment by uesteibar** [comment]
I would add the modal in every step change, and not disable the button ever, I think that's the less restrictive and simplest thing we can do. Maybe some teams don't want to vote?

I don't think the current checks prevent the case I explained above, because if someone else voted, what's stopping you from accidentally jumping to the next phase? A modal here would prevent that.

Let's see what others think here too :)

**Comment by rudeayelo** [comment]
That will add two more clicks to every retro, maybe we could add those modals conditionally:

- don't disable the buttons (but the "Start retro" one, if no items we still disable it)
- **only if no votes are casted** and the "Add action items" button is clicked, show the modal (similar to the mockup in my previous comment) asking to confirm if you want to skip voting
- **only if no action items were added** and the "Finish retro" button is clicked, show the modal in my previous comment

**Comment by uesteibar** [comment]
I repeat, that wouldn't prevent the case I'm talking about above from happening. If we change something on how jumping steps work, let's cover that case please 🙏 

For the record: I'm fine with also not changing anything, that's why I filed this as a _discussion_.

**Comment by rudeayelo** [comment]
Oops sorry I totally skipped the second paragraph.

Maybe too much of an edge case?

**Comment by uesteibar** [comment]
I don't know what happened, maybe someone else clicked by accident? Still, it might be pointing out that it's _far too easy_ to jump to the next step, specially for an irreversible action.

**Comment by atrzaska** [comment]
We should think about adding feedback button on the tool. The theory is that not everyone understands the steps correctly. Are we fine with not everyone understanding the steps  correctly? How much % of users don't understand the steps?

Personal opinion:

The tool took a bit to understand at first. I think we can make some copy improvements, or UI improvements.

For example, IMO the text "Add your comments below, you won't be able to see your peers' until next step" is very hard to see. It's small and barely visible. Maybe we could increase visibility with more contrast?

Having a previous step feels wrong for me. Maybe it's the feeling of going back instead of progressing forward.

**Comment by uesteibar** [comment]
> For example, IMO the text "Add your comments below, you won't be able to see your peers' until next step" is very hard to see. It's small and barely visible. Maybe we could increase visibility with more contrast?

➕💯 to this

> Having a previous step feels wrong for me.

I'm OK with not being able to go back, I agree it might complicate things. What do you think of adding some gatekeeping with modals?

**Comment by atrzaska** [comment]
@uesteibar I think modals should be OK. Having a modal will allow us to have a place to explain each step also.

**Comment by marciobarrios** [comment]
I also agree on giving more visibility to "instructions" as a first step, and also to add a tooltip to the button on hover to explain that the action is irreversible. But I would wait to add extra modals for confirmation, because maybe it's enough with better and more visible instructions.

Do we have a plan? 😊

**Comment by uesteibar** [comment]
For irreversible actions, I prefer something more blocking like a modal rather than something cosmetic like a tooltip. The former we are sure that users _see_ before actually jumping phases, the later a user might not see at all.

**Comment by Thra11** [comment]
When we used retro tool for the first time a few weeks ago, we had some misunderstandings. I think the UI made it fairly clear that advancing to the next step was irreversible. What people didn't immediately understand was that they were advancing the entire board for everyone, not just their own view of the board. I think many people thought that once they'd added their comments, they should click the blue button to share their comments with everyone else and start reading other people's comments, assuming that other people who were still writing comments would stay on the first step until they click their own button.

We ended up having to create a new board and start all over again. If you don't intend to allow people to go back, perhaps you need to make it more obvious not just that it's irreversible, but also that it will advance the global state of the board, not just the individual state of the user.
